### PR TITLE
fix(reconciliation): fail-closed stop-loss lookup — never re-place a stop on an unconfirmed lookup (#713)

### LINKED ISSUE #713: Reconciler treats get_order()=None (transient API failure) as 'missing stop' -> can place a DUPLICATE stop
Found during #710 review (codex). `BinanceProvider.get_order` returns `None` on **all** exceptions (binance_provider.py ~1421). The periodic reconciler treats `sl_order is None` as 'stop missing' and re-places one (reconciliation.py ~2754). A **transient API failure** (timeout/rate-limit) therefore can create a **duplicate** SELL stop (long) or BUY stop (short) — which reserves base/margin and can itself trigger -2010 on a later close (cf. #710).

**Fix direction:** distinguish 'confirmed not found' from 'unknown / API failure'. Use a fail-closed accessor (return a distinct sentinel on lookup failure, like `has_open_orders`) and treat 'unknown' as 'do not re-place this cycle' rather than 'missing'. Add a test for transient-failure → no duplicate.

Pre-existing; separated from #710 per scoping decision. Raises in importance as the reconciler becomes a more central re-protection path (#710).

### PR DESCRIPTION
**Pull request by alexflorisca**
## Summary

Fixes #713: the reconciler treated `get_order() → None` as "stop-loss missing" and re-placed a stop — but `BinanceProvider.get_order` returns `None` on **every** exception, so a transient API failure (timeout, rate limit, connectivity blip) could create a **duplicate** protective stop while the original still rests on the exchange. A duplicate stop reserves base/margin (→ later `-2010` rejections on the close path, cf. #710) and can flip the position if both stops fill.

Both decision sites had the bug (same class, LESSONS §1.8 / §2.1):
- `PositionReconciler._verify_stop_loss` (startup recovery)
- `PeriodicReconciler` stop-verification loop (every ~120 s)

## Changes

- **`ExchangeInterface.get_order_checked` + `OrderLookupError`** — fail-closed lookup contract: returns the `Order`; returns `None` **only** when the exchange confirmed the order does not exist; raises `OrderLookupError` when the lookup cannot be confirmed.
- **`BinanceProvider.get_order_checked`** — `None` only on Binance `-2013` ("Order does not exist"); any other API error, network error, unavailable client, or unparseable response raises `OrderLookupError`. Numeric IDs route via `orderId`, alphanumeric client IDs via `origClientOrderId`.
- **Reconciler** — new `lookup_order_fail_closed()` helper used by both verifiers: on an unconfirmed lookup it logs a warning and **skips the position for this cycle** (tracked `stop_loss_order_id` retained, no MISSING audit, no re-placement). Confirmed-missing stops are still re-placed exactly as before. Falls back to legacy `get_order` for providers without the accessor (unchanged behavior).

Audited the other four `get_order` call sites (`reconciliation.py` entry verify ×2, `trading_engine._stop_loss_filled_quantity`, `_check_stop_loss_filled`, legacy SL reconcile): all already fail-safe on `None` (no state mutation / conservative default) — left unchanged.

## Testing

- New: `TestGetOrderChecked` (6 provider tests: found / `-2013`→`None` / API error→raise / network→raise / client-unavailable→raise / client-id routing) and `TestFailClosedSLLookup` (periodic skip-on-unconfirmed, startup skip-on-unconfirmed, regression guard: confirmed-missing still re-places).
- Existing reconciliation suites pass unchanged in semantics via a fixture delegation (`get_order_checked` → `get_order`), so all prior order-driving patterns keep working.
- Full unit suite: **3869 passed, 103 skipped** locally.

## Risk

No behavior change when lookups succeed. The only new behavior: a failed lookup defers action to the next reconcile cycle (~2 min) instead of immediately re-placing — strictly safer for a protective-order decision.

https://claude.ai/code/session_01Atv7dvZtsdpkPJhZdi6QMN

---
_Generated by [Claude Code](https://claude.ai/code/session_01Atv7dvZtsdpkPJhZdi6QMN)_

### CODE REVIEW

File: src/engines/live/reconciliation.py [hunk 1]
```diff
@@ -46,6 +46,36 @@
 logger = logging.getLogger(__name__)
 
 
+def lookup_order_fail_closed(exchange: Any, order_id: str, symbol: str) -> tuple[Any, bool]:
```
[the remark this answers is not in the document]
Comment (alexflorisca) [reply src/engines/live/reconciliation.py]: Done in 66648f1 — added a `_OrderLookup` Protocol and typed the return as `tuple[Order | None, bool]`. One deviation from the snippet: the Protocol declares only `get_order` (the guaranteed member). `get_order_checked` is intentionally left out because a Protocol can't express an optional member — it stays a `getattr` probe, documented in the Protocol docstring.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/data_providers/exchange_interface.py [hunk 2]
```diff
@@ -189,6 +199,22 @@ def get_order(self, order_id: str, symbol: str) -> Order | None:
...
+        :class:`OrderLookupError` when the lookup could not be confirmed
+        (network error, rate limit, API failure).
+
+        The default delegates to :meth:`get_order`, which for most providers
+        swallows errors into ``None`` — i.e. the default CANNOT distinguish
+        "confirmed absent" from "unknown". Providers used for live trading
+        MUST override this with an implementation that raises
+        ``OrderLookupError`` on unconfirmed lookups (see ``BinanceProvider``).
+        """
+        return self.get_order(order_id, symbol)
```
[the remark this answers is not in the document]
Comment (alexflorisca) [reply src/data_providers/exchange_interface.py]: Done in c429275 — took both halves of the suggestion: the ABC default now raises `OrderLookupError` (fail-closed for any provider without an override), AND `CoinbaseProvider` got a real override (`None` only on a confirmed HTTP 404; everything else raises). Tests added for both.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/engines/live/reconciliation.py [hunk 3]
```diff
@@ -34,18 +34,62 @@
...
 if TYPE_CHECKING:
-    from src.data_providers.exchange_interface import ExchangeInterface
+    from src.data_providers.exchange_interface import ExchangeInterface, Order
     from src.database.manager import DatabaseManager
     from src.engines.live.execution.position_tracker import LivePositionTracker
 
 logger = logging.getLogger(__name__)
 
 
+@runtime_checkable
```
[the remark this answers is not in the document]
Comment (alexflorisca) [reply src/engines/live/reconciliation.py]: Done in c429275 — dropped `@runtime_checkable` (the import stays; `_HasDbPositionId` still uses it).

---
_Generated by [Claude Code](https://claude.ai/code)_
